source/index.ts

```diff
@@ -109,49 +104,49 @@ function is(value: any): TypeName { // tslint:disable-line:only-arrow-functions
 }
 
 namespace is { // tslint:disable-line:no-namespace
-	const isObject = (value: any) => typeof value === 'object';
+	const isObject = (value: object) => typeof value === 'object'; // tslint:disable-line:strict-type-predicates
 
 	// tslint:disable:variable-name
 	export const undefined = isOfType<undefined>('undefined');
 	export const string = isOfType<string>('string');
 	export const number = isOfType<number>('number');
 	export const function_ = isOfType<Function>('function');
-	export const null_ = (value: any): value is null => value === null;
+	export const null_ = (value: null | any): value is null => value === null;
 	export const class_ = (value: any) => function_(value) && value.toString().startsWith('class ');
-	export const boolean = (value: any): value is boolean => value === true || value === false;
+	export const boolean = (value: boolean): value is boolean => value === true || value === false;
 	export const symbol = isOfType<Symbol>('symbol');
 	// tslint:enable:variable-name
 
 	export const array = Array.isArray;
 	export const buffer = Buffer.isBuffer;
 
-	export const nullOrUndefined = (value: any): value is null | undefined => null_(value) || undefined(value);
-	export const object = (value: any) => !nullOrUndefined(value) && (function_(value) || isObject(value));
-	export const iterable = (value: any): value is Iterator<any> => !nullOrUndefined(value) && function_(value[Symbol.iterator]);
-	export const generator = (value: any): value is Generator => iterable(value) && function_(value.next) && function_(value.throw);
+	export const nullOrUndefined = (value: null | undefined | any): value is null | undefined => null_(value) || undefined(value);
+	export const object = (value: object) => !nullOrUndefined(value) && (function_(value) || isObject(value));
+	export const iterable = <T = any>(value: IterableIterator<T>): value is IterableIterator<T> => !nullOrUndefined(value) && function_(value[Symbol.iterator]);
```

Comment:
We discussed this and we decided not to do this. I explain [here](https://github.com/sindresorhus/is/pull/43#issuecomment-377780830) why.

> I don't think we should do that. The only thing that is does, is check the type and infer the type correctly. It's not up to is to do more than that. If users type there values correctly, all of it will go automatically.